make/compiler/Makefile.gnu

```diff
@@ -187,6 +187,16 @@ RUNTIME_CFLAGS += -Wno-stringop-overflow
 SQUASH_WARN_GEN_CFLAGS += -Wno-array-bounds
 endif
 
+#
+# Avoid errors about insufficient initializer lifetimes because they
+# occur in LLVM headers.  We would like to know when this occurs,
+# though, so don't turn off the warning; just don't let it abort the
+# build.
+#
+ifeq ($(shell test $(GNU_GPP_MAJOR_VERSION) -eq 9; echo "$$?"),0)
```

Comment:
Just for 9 or should it be 9 or newer?

Reply:
We added some warnings just for 8, and they are no longer needed in 9, so I figured we'd try the same strategy again.

Reply:
Sounds good